### PR TITLE
[link] fix: anchors issues

### PR DESCRIPTION
**Pull request by maimonav**

### CODE REVIEW

File: packages/viewer/web/src/getPluginViewers.jsx [hunk 1]
```diff
@@ -10,6 +10,7 @@ import {
   normalizeUrl,
   IMAGE_TYPE,
   GALLERY_TYPE,
+  isSSR,
```
Comment (maimonav) [inline packages/viewer/web/src/getPluginViewers.jsx:13]:
```suggestion
  isSSR,
```

File: packages/plugin-image/web/src/image-viewer.jsx [hunk 2]
```diff
@@ -260,8 +260,7 @@ class ImageViewer extends React.Component {
     if (this.hasLink()) {
       return null;
     } else if (this.hasAnchor()) {
-      e.preventDefault();
-      this.scrollToAnchor();
+      this.scrollToAnchor(e);
```
Comment (shaulgo) [inline packages/plugin-image/web/src/image-viewer.jsx:263]: no need to pass event through 3 functions.

File: packages/viewer/web/src/getPluginViewers.jsx [hunk 3]
```diff
@@ -80,6 +80,12 @@ class PluginViewer extends PureComponent {
     );
   };
 
+  getUrlWithAnchorTag = anchorString => {
```
Comment (shaulgo) [inline packages/viewer/web/src/getPluginViewers.jsx:83]: is this used? 
Also duplicate.

File: packages/plugin-link/web/src/toolbar/UrlLinkButton.jsx [hunk 4]
```diff
@@ -12,14 +12,14 @@ export default class UrlLinkButton extends Component {
...
-  handleClick = () => {
+  handleClick = e => {
     const { getEditorState } = this.props;
     const linkData = getLinkDataInSelection(getEditorState());
     const { anchor = '' } = linkData || {};
     const nodeListOfAllblocks = document.querySelectorAll(`[data-editor]`);
     const arrayOfAllblocks = Array.apply(null, nodeListOfAllblocks);
     const element = arrayOfAllblocks.find(block => block.dataset.offsetKey === `${anchor}-0-0`);
-    anchorScroll(element);
+    anchorScroll(e, element);
```
Comment (shaulgo) [inline packages/plugin-link/web/src/toolbar/UrlLinkButton.jsx:22]: Was there need to preventDefault here? What's UrlLinkButton?

File: packages/plugin-link/web/src/LinkViewer.jsx [hunk 5]
```diff
@@ -34,29 +34,27 @@ class LinkViewer extends Component {
...
     }
   };
 
   getHref(url, anchor) {
-    const siteUrl = this.props.config?.[LINK_TYPE]?.siteUrl;
     if (url) {
       return normalizeUrl(url);
-    } else if (siteUrl) {
-      return `${siteUrl}#viewer-${anchor}`;
+    }
```
Comment (shaulgo) [inline packages/plugin-link/web/src/LinkViewer.jsx:52]: Will this work?
`return url ? normalizeUrl(url) : `#viewer-${anchor}``

**Review by shaulgo — CHANGES_REQUESTED** [review]

File: packages/viewer/web/src/getPluginViewers.jsx (hunk 3 above)
Comment (maimonav) [reply packages/viewer/web/src/getPluginViewers.jsx:83]: Oops, it should have been deleted.

File: packages/plugin-link/web/src/LinkViewer.jsx (hunk 5 above)
Comment (maimonav) [reply packages/plugin-link/web/src/LinkViewer.jsx:52]: I think so, as long as we have at least one of (url,anchor) defined.

File: packages/common/web/src/Utils/anchor.ts [hunk 6]
```diff
@@ -7,3 +7,9 @@ export const anchorScroll = element => {
   element.style.marginTop = null;
```
Comment (shaulgo) [inline packages/common/web/src/Utils/anchor.ts:7]: const {marginTop, paddingTop} = element.style;

**Review by shaulgo — APPROVED** [review]

**Pull request merged**